packages/react-compass/src/utils/objectToCss/object-to-css.ts

```diff
@@ -0,0 +1,65 @@
+import theme from '../../themeV2/theme'
+export interface StyleObject {
+  [key: string]: string | StyleObject
+}
+function handleVariables<T>(
+  obj: Record<string, unknown>,
+  targetKey: string,
+): T | undefined {
+  const targetKeyWithoutPrefix = targetKey.startsWith('$')
+    ? targetKey.slice(1)
+    : targetKey
+  for (const key in obj) {
+    if (key in obj) {

```

Comment:
Is it really necessary to check if the object has the key while iterating over the object?